interview-questions.md

```diff
@@ -8,50 +8,54 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: A parameter is a variable or a placeholder used in the function/method definition to represent a value that the function expects to receive. An argument is a specific value or expression passed to a function/method when it is called or invoked.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: It takes the value of each index in the array been map. 
+
+Researched answer: For a .map() it takes a callback function as its parameter, which is called for each element in the array:
+           currentValue (required): The current element being processed in the array.
+           index (optional): The index of the current element being processed in the array.
+           array (optional): The array on which the .map() method was called.
 
-Researched answer:
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: .map method will iterarte throught the array and output the array with the same lenght. the .filter method iterates throught the array but output an array with different lenght due to the condition set in method. Only the value that passed the conditional will pass to the new array.
 
-Researched answer:
+Researched answer: They both take iterate throught an array and output a new on. A map outputs a new array with same lenght as the original while the .filter outputs a different lenght.
 
 1. What is iteration?
 
-Your answer:
+Your answer: the repetition of task in programming
 
-Researched answer:
+Researched answer: Iteration refers to the process of repeatedly performing a set of instructions or operations. In programming, iteration is often used to perform tasks on a collection of elements, such as iterating over an array or a list, to process each element individually.
 
 1. What is the difference between a function and a method?
 
-Your answer:
+Your answer: a funtion is a block of code created by the programmer that can be invoked many time. A method is a built in function in the proggraming language.
 
-Researched answer:
+Researched answer: A function is a block of code that performs a specific task or set of instructions. It is a self-contained unit of code that can be called or invoked from different parts of a program. Functions are generally standalone and not associated with any particular object or class. A method is a function that is associated with an object or a class. It is a behavior or action that an object or class can perform.
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: The ability to access a variable within a self contain block of code.
 
-Researched answer:
+Researched answer: s a JavaScript behavior where variable and function declarations are moved to the top of their containing scope during the compilation phase, while still maintaining their original order in the code. 
```

Comment:
A lot of these researched answers are too close to being copied and pasted. From now on, make sure you are actually learning the concept and then answering these completely in your own words.